mso/resource_mso_schema_template_service_graph.go

```diff
@@ -816,3 +520,40 @@ func getNodeNameFromId(msoClient *client.Client, nodeId string) (string, error)
 
 	return "", fmt.Errorf("Unable to find nodeNamefor nodeid %s", nodeId)
 }
+
+func getServiceGraphNodes(d *schema.ResourceData, msoClient *client.Client) ([]interface{}, error) {
+	serviceNodes := make([]interface{}, 0, 1)
+	if tempVar, ok := d.GetOk("service_node_type"); ok {
+		serviceNodeType := tempVar.(string)
+		nodeId, err := getNodeIdFromName(msoClient, serviceNodeType)
+		if err != nil {
+			return nil, err
+		}
+		serviceNode := map[string]interface{}{
+			"name":              "node1",
+			"index":             1,
+			"serviceNodeTypeId": nodeId,
+		}
+		serviceNodes = append(serviceNodes, serviceNode)
+	} else {
+		if val, ok := d.GetOk("service_node"); ok {
+			for i, val := range val.([]interface{}) {
+				serviceNodeValues := val.(map[string]interface{})
+				if serviceNodeValues["type"] != "" {
+					nodeId, err := getNodeIdFromName(msoClient, fmt.Sprintf("%v", serviceNodeValues["type"]))
```

Comment:
Here we execute the same api call `cont, err := msoClient.GetViaURL("api/v1/schemas/service-node-types")` multiple times for each service node, where I think we should only do the api call once and return the types. Then have the type lookup in separate function for each service node.